plugins/snowflake/dbt/include/snowflake/macros/materializations/merge.sql

```diff
@@ -24,5 +25,20 @@
         {{ default__get_merge_sql(target, source_sql, unique_key, dest_columns, predicates) }}
 
     {%- endif -%}
+    {%- endset -%}
+    
+    {% do return(snowflake_dml_explicit_transaction(dml)) %}
 
 {% endmacro %}
+
+
+{% macro snowflake__get_delete_insert_merge_sql(target, source, unique_key, dest_columns) %}
+    {% set dml = default__get_delete_insert_merge_sql(target, source, unique_key, dest_columns) %}
+    {% do return(snowflake_dml_explicit_transaction(dml)) %}
```

Comment:
is there any problem with specifying code with an explicit `begin/commit` in addition to enabling autocommit? Surprised to see that this works! I would have expected that we would need to disable autocommit for these queries

Reply:
This is one of the things that's tricky about `autocommit`. You don't actually need to disable it, it just doesn't take effect if you supply explicit `begin` + `commit`. From [docs](https://docs.snowflake.com/en/sql-reference/transactions.html#autocommit):

> Statements inside an explicit transaction are not affected by AUTOCOMMIT. For example, statements inside an explicit BEGIN TRANSACTION ... ROLLBACK are rolled back even if AUTOCOMMIT is TRUE.